src/Picasso_BatchedLinearAlgebra.hpp

```diff
@@ -1525,8 +1521,7 @@ KOKKOS_INLINE_FUNCTION auto diagonal( const ExpressionX& x )
                                   ExpressionX::extent_0, ExpressionX::extent_0>(
         [=]( const int i, const int j ) {
             return ( i == j ) ? x( i )
-                              : Kokkos::ArithTraits<
-                                    typename ExpressionX::value_type>::zero();
+                              : 0.0;
```

Comment:
I doubt that would work for non floating point

Reply:
I don't expect to do integer arithmetic with this class